### PR TITLE
Support `CROISSANT_GIT_USERNAME`/`CROISSANT_GIT_PASSWORD`

### PR DESCRIPTION
**Pull request by marcenacp**

### CODE REVIEW

**Review by ccl-core — APPROVED** [review]
Thanks!

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/download.py [hunk 1]
```diff
@@ -59,6 +59,27 @@ def get_download_filepath(node: Node, url: str) -> epath.Path:
     return constants.DOWNLOAD_PATH / f"croissant-{hashed_url}"
 
 
+def insert_credentials(url: str, username: str | None, password: str | None) -> str:
+    """Inserts credentials in the URL for an HTTP authentication.
+
+    Example: https://{username}:{password}@github.com/account/repo
```
Comment (ccl-core) [inline python/ml_croissant/ml_croissant/_src/operation_graph/operations/download.py:65]: Maybe it should be

```
Returns: the URL populated with the given credentials. For example: `https://{username}...`

```

Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/download.py:65]: Very good point! I fixed this in the next PR.

Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/download.py:65]: (Follow-up: https://github.com/mlcommons/croissant/pull/180)

**Pull request merged**